Sources/LoopAlgorithm/LoopQuantity.swift

```diff
@@ -0,0 +1,69 @@
+//
+//  File.swift
+//  LoopAlgorithm
+//
+//  Created by Cameron Ingham on 11/8/24.
+//
+
+import Foundation
+
+public struct LoopQuantity: Hashable, Equatable, Comparable, Sendable {
+
+    public let unit: LoopUnit
+    public let value: Double
+    
+    public init(unit: LoopUnit, doubleValue value: Double) {
+        self.unit = unit
+        self.value = value
+    }
+    
+    public func `is`(compatibleWith unit: LoopUnit) -> Bool {
+        self.unit.conversionFactor(from: unit) != nil
+    }
+    
+    /**
+     @method        doubleValueForUnit:
+     @abstract      Returns the quantity value converted to the given unit.
+     @discussion    Throws an exception if the receiver's value cannot be converted to one of the requested unit.
+     */
+    public func doubleValue(for unit: LoopUnit) -> Double {
+        guard let conversionFactor = self.unit.conversionFactor(from: unit) else {
+            fatalError("Conversion Error: \(self.unit.unitString) is not compatible with \(unit.unitString).")
```

Comment:
Could this be different error handling?

`enum LoopQuantityError: Error {
    case incompatibleUnits(from: LoopUnit, to: LoopUnit)
}`

Reply:
In an attempt to match the existing HealthKit API at the call-sites, this function cannot throw an error to be handled down the line. While I like the idea of having some better error handling here that would require a very large change throughout the entire codebase well beyond what's already required with this update. I could certainly take a look at moving towards this but it might be better to do so incrementally.